### PR TITLE
Remove one layer in the compilation process

### PR DESCRIPTION
**Pull request by ltratt**
Previously the yksom compiler produced `cobject`'s, a post-AST, but pre-run-time data structure that the interpreter than converted into its internal format and ran. However, this hasn't turned out to be a particularly useful layer: `cobject`'s are almost identical in structure to run-time `object`'s (3 out of the 4 `cobject`'s are literally identical). This PR thus removes this intermediate layer entirely: the compiler now creates run-time values directly.

There is probably a tiny performance benefit from doing this, but it's almost certainly too small to measure. The advantage of this PR is mostly code simplification (100 lines fewer code overall!). However, it also makes one future thing easier: SOM dynamically loads classes and previously this felt slightly awkward due to the `cobject` layer. By removing that layer, it should be more obvious how to do dynamic class loading in the future.

### CODE REVIEW

**Comment by ltratt** [comment]
@ummarikar You might want to have a look at this one and offer comments (though @ptersilie is the main reviewer).

**Comment by ltratt** [comment]
bors try

**Comment by ummarikar** [comment]
With my current understanding, everything seems fine. But, I will need to go deeper into the workings of yksom to provide value in future PRs!

File: src/lib/compiler/cobjects.rs [hunk 1]
```diff
@@ -11,24 +11,16 @@ use std::path::PathBuf;
 
 use crate::compiler::instrs::Instr;
 
-pub use crate::vm::objects::{Method, MethodBody};
+pub use crate::vm::objects::{BlockInfo, Method, MethodBody};
```
Comment (ptersilie) [inline src/lib/compiler/cobjects.rs]: The commit message says `vm::objects::class::BlockInfo` but this imports ``vm::objects::BlockInfo``. I presume the latter is correct?

Comment (ltratt) [reply src/lib/compiler/cobjects.rs]: They both point to the same thing underneath (the latter is shorthand for the former).

Comment (ptersilie) [reply src/lib/compiler/cobjects.rs]: Nevermind, just saw that `objects` imports `BlockInfo`.

**Comment by ptersilie** [comment]
Looks fine.

bors r+

**Comment by ptersilie** [comment]
It seems nothing has changed since the last commit, yet the build still fails. Odd.

**Comment by ltratt** [comment]
It's because #73 was merged after this PR was raised and that PR references the old layout. I need to rebase against master to fix this (it's a one line fix but I can't push it without force pushing). OK?

**Comment by ptersilie** [comment]
Go ahead!

**Comment by ltratt** [comment]
I meant https://github.com/softdevteam/yksom/pull/76 (which was raised after this PR, but was merged first).

**Comment by ltratt** [comment]
It's hard to see but one line (237) in https://github.com/softdevteam/yksom/pull/75/commits/dabe52e6ac10733909eaedd63d628f65afc4dc05 has now changed. I think this is ready to try again.

**Pull request merged**